compose/volume.py

```diff
@@ -97,28 +97,32 @@ def initialize(self):
                             )
                         )
                     continue
-                log.info(
-                    'Creating volume "{0}" with {1} driver'.format(
-                        volume.full_name, volume.driver or 'default'
+
+                if not volume_exists:
+                    log.info(
+                        'Creating volume "{0}" with {1} driver'.format(
+                            volume.full_name, volume.driver or 'default'
+                        )
                     )
-                )
-                volume.create()
+                    volume.create()
+                else:
+                    driver = volume.inspect()['Driver']
+                    if driver != volume.driver:
```

Comment:
Won't this fail in the case where no driver is specified, since `local != None` ?